activesupport/lib/active_support/configuration_file.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module ActiveSupport
+  class ConfigurationFile
+    INVISIBILE_WHITESPACE = /\U+A0/
+    # Reads a YAML file that might have ERB syntax, evaluates the ERB, and then
+    # parses the resulting YAML.
+    #
+    # If there are characters that will confuse YAML (Like invisible
+    # non-breaking spaces) it will warn you.
+    def initialize(content:, context: nil)
+      @content = content
+      @context = context
+    end
+
+    attr_reader :content
+    attr_reader :context
+    
+    def self.read(content:, context: nil)
+      new(content:, context: nil).read
+    end
+
+    def read
+      if content.include?(INVISIBLE_WHITESPACE)
+        warn "File contains invisible non-breaking spaces, you may want to remove those"
+      end
+      template = ERB.new(content)
+
+      yaml = if context
+        template.result(context)
+      else
+        template.result
+      end
+
+      YAML.safe_load(yaml) || {}
+    rescue ArgumentError, Psych::SyntaxError => error
+      raise Fixture::FormatError, "a YAML error occurred parsing #{path}. Please note that YAML must be consistently indented using spaces. Tabs are not allowed. Please have a look at https://www.yaml.org/faq.html\nThe exact error was:\n  #{error.class}: #{error}", error.backtrace
```

Comment:
Can't use `Fixture::FormatError` here, it's defined within Active Record. Rescuing `ArgumentError` here seems fraught as well, if we remove that do we not to capture the `Psych::SyntaxError` at all? I mean, won't it print a warning method that's better than ours here?

Reply:
Good question, and no it has a terrible error message (see original issue).

I'm surprised the FormatError is limited to activerecord? Maybe it shouldn't?

Reply:
It's because `Fixture` is defined in Active Record (as it should). I'm fine with extracting a `ConfigurationFile::FormatError`

Reply:
Handled.